Update README and RF docs for 0.8 (#730)
rapidsai/cuml: README.md

cuML is a suite of libraries that implement machine learning algorithms and mathematical primitives functions that share compatible APIs with other [RAPIDS](https://rapids.ai/) projects.

cuML enables data scientists, researchers, and software engineers to run
traditional tabular ML tasks on GPUs without going into the details of CUDA
programming. In most cases, cuML's Python API matches the API from
[scikit-learn](https://scikit-learn.org).

For large datasets, these GPU-based implementations can complete 10-50x faster
than their CPU equivalents. For details on performance, see the [cuML Benchmarks

> Notebook](https://github.com/rapidsai/notebooks-extended/blob/master/intermediate_notebooks/benchmarks/cuml_benchmarks.ipynb).

Reviewer: This is great for the front page!

As an example, the following Python snippet loads input and computes DBSCAN clusters, all on GPU:
```python
import cudf

> from cuml.cluster import DBSCAN

Reviewer 2: ```suggestion
from cuml import DBSCAN
```

The .cluster is not needed, so we want to change the default example here?

Reviewer 2: This is more of a question than a change request

Author: To me, it was nice to match what sklearn does, since you would do from sklearn.cluster import DBSCAN and not from their base namespace. If you prefer the old way, I can live with that though.

dtype: int32
```

For additional examples, browse our complete [API
documentation](https://docs.rapids.ai/api/cuml/stable/), or check out our more
detailed [walkthrough
notebooks](https://github.com/rapidsai/notebooks/tree/master/cuml).

> 

Reviewer: Should we add a link to the notebooks-extended repository as well for completeness?

### Supported Algorithms
| Category | Algorithm | Notes |
| --- | --- | --- |

> | **Clustering** |  Density-Based Spatial Clustering of Applications with Noise (DBSCAN) | |

Reviewer: Glad we're now breaking this down by category

|  | K-Means | |
|  | K-Nearest Neighbors (KNN) | Multi-GPU with [dask-cuml](http://github.com/rapidsai/dask-cuml) <br> Uses [Faiss](https://github.com/facebookresearch/faiss) |
| **Dimensionality Reduction** | Principal Components Analysis (PCA) | |
| | Truncated Singular Value Decomposition (tSVD) | Multi-GPU version available (CUDA 10 only) |
| | Uniform Manifold Approximation and Projection (UMAP) | |
| | Random Projection | |
| **Linear Models for Regression or Classification** | Linear Regression (OLS) | Multi-GPU available in conda CUDA 10 package and [dask-cuml](http://github.com/rapidsai/dask-cuml) |

> | | Linear Regression with Lasso or Ridge Regularization | |

Reviewer 2: ```suggestion
| | Linear Regression with Lasso or Ridge Regularization | Ordinary Linear Regression Multi-GPU  with [dask-cuml](http://github.com/rapidsai/dask-cuml)|
```

| | ElasticNet Regression | |
| | Logistic Regression | |

> | | SGD, LBFGS, and Coordinate Descent Solvers for linear models | |

Reviewer 3: it is a very small thing, i think having the expanded form of SGD and LBFGS might help the readers

Reviewer 2: For consistency indeed they should be 

```suggestion
| | Stochastic Gradient Descent (SGD), Quasi-Newton (QN) including L-BFGS and OWL-QN, and Coordinate Descent (CD) Solvers for linear models | |
```
